docs/CRUD.md

```diff
@@ -0,0 +1,470 @@
+    meta, // Additional information about the list, like facets & statistics
+    isPending, // Boolean, true until the data is available
+    isFetching, // Boolean, true while the data is being fetched, false once the data is fetched
+    isLoading, // Boolean, true until the data is fetched for the first time
+
+    // Pagination
+    page, // Current page. Starts at 1
+    perPage, // Number of results per page. Defaults to 25
+    setPage, // Callback to change the page, e.g. setPage(3)
+    setPerPage, // Callback to change the number of results per page, e.g. setPerPage(25)
+    hasPreviousPage, // Boolean, true if the current page is not the first one
+    hasNextPage, // Boolean, true if the current page is not the last one
+    
+    // Sorting
+    sort, // Sort object { field, order }, e.g. { field: 'date', order: 'DESC' }
+    setSort, // Callback to change the sort, e.g. setSort({ field: 'name', order: 'ASC' })
+    
+    // Filtering
+    filterValues, // Dictionary of filter values, e.g. { title: 'lorem', nationality: 'fr' }
+    displayedFilters, // Dictionary of displayed filters, e.g. { title: true, nationality: true }
+    setFilters, // Callback to update the filters, e.g. setFilters(filters, displayedFilters)
+    showFilter, // Callback to show one of the filters, e.g. showFilter('title', defaultValue)
+    hideFilter, // Callback to hide one of the filters, e.g. hideFilter('title')
+    
+    // Record selection
+    selectedIds, // Array listing the ids of the selected records, e.g. [123, 456]
+    onSelect, // Callback to change the list of selected records, e.g. onSelect([456, 789])
+    onToggleItem, // Callback to toggle the record selection for a given id, e.g. onToggleItem(456)
+    onUnselectItems, // Callback to clear the record selection, e.g. onUnselectItems();
+    
+    // Misc
+    defaultTitle, // Translated title based on the resource, e.g. 'Posts'
+    resource, // Resource name, deduced from the location. e.g. 'posts'
+    refetch, // Callback for fetching the list data again
+} = listContext;
+```
+
+## CRUD Routes
+
+You could declare the CRUD routes manually using react-router's `<Route>` component. But it's such a typical pattern that react-admin provides a shortcut: the [`<Resource>`](./Resource.md) component.
+
+```jsx
+<Resource
+    name="posts"
+    list={PostList}     // maps PostList to /posts
+    show={PostShow}     // maps PostShow to /posts/:id/show
+    edit={PostEdit}     // maps PostEdit to /posts/:id
+    create={PostCreate} // maps PostCreate to /posts/create
+/>
+```
+
+This is the equivalent of the following react-router configuration:
+
+```jsx
+<ResourceContextProvider value="posts">
+    <Route path="/posts" element={<PostList />} />
+    <Route path="/posts/:id/show" element={<PostShow />} />
+    <Route path="/posts/:id" element={<PostEdit />} />
+    <Route path="/posts/create" element={<PostCreate />} />
+</ResourceContextProvider>
```

Comment:
I don't think this is valid react-router code as you must use `Route` as a direct child of either a `Routes` or `Route` element (See https://github.com/remix-run/react-router/blob/main/packages/react-router/lib/components.tsx#L676-L682)